Added allow_redirects for http monitor

kz159 opened the pull request:
Hi there! i've added ability to not follow redirects
But i've got couple of questions:
1) is it normal that i've got this on `make unit-test`
```
CRITICAL simplemonitor.logger-unnamed:file.py:222 output folder html does not exist
CRITICAL simplemonitor.logger-unnamed:file.py:424 Target folder html does not exist
============================================================== short test summary info ===============================================================
FAILED tests/test_alerter.py::TestAlerter::test_should_alert_catchup - AssertionError: <AlertType.NONE: 'none'> != <AlertType.CATCHUP: 'catchup'>
FAILED tests/test_alerter.py::TestAlerter::test_should_alert_limit - AssertionError: <AlertType.NONE: 'none'> != <AlertType.FAILURE: 'failure'>
FAILED tests/test_alerter.py::TestAlerter::test_should_alert_no_catchup - AssertionError: <AlertType.NONE: 'none'> != <AlertType.FAILURE: 'failure'>
FAILED tests/test_alerter.py::TestAlerter::test_should_alert_ooh - AssertionError: <AlertType.NONE: 'none'> != <AlertType.FAILURE: 'failure'>
FAILED tests/test_logger.py::TestHTMLLogger::test_html - AssertionError: '<!DOCTYPE html>\n<html lang="en">\n  <hea[5369 chars]l>\n' != ''
FAILED tests/test_logger.py::TestHTMLLogger::test_html_map - AssertionError: '<!DOCTYPE html>\n<html lang="en">\n  <hea[3938 chars]l>\n' != ''
FAILED tests/test_logger.py::TestHTMLLogger::test_html_tz - AssertionError: '<!DOCTYPE html>\n<html lang="en">\n  <hea[5369 chars]l>\n' != ''
```

2) I didn't understand how network test are made and for sake of perfectionism i wanna test this feature, can you help me?

Also happy new year and thanks for this small yet usefull software

kz159 commented:
Will add this to documentation after i`ll get approve

jamesoff commented:
Hey :) Thanks for the PR!

> is it normal that i've got this on make unit-test

Unit tests *should* be passing at the moment, although I see a couple of warnings showing up now about some deprecations in dependencies. (That's on Python 3.11.0.)

The HTML-related errors probably just need you to `mkdir html` in your checked-out version; it's not included in the repo but is the target for the rendering of the HTML pages during tests. (The Github workflow includes the `mkdir` before it runs the tests, if you want to compare.)

Not sure why the AlertType ones are failing, I don't see anything in your changes which would affect that. Did you branch off of recently `develop`? I'll approve the workflow to run and we can see if it passes on Github :)

> how network test are made

Assuming you mean how to test monitors which use the network (i.e. like this one) :) at the moment, test coverage for that is poor. I think the best approach is to mock `requests` so that you can (in the test) confirm that `requests.get()` was invoked in the right parameters (and it doesn't actually need to go out on to the network). (I was learning how to write unit tests when I started adding them to this project, which is why they're not great and there's poor coverage for things like that.)

If you're happy with mocking requsts (with `unittest.mock`) then go ahead and do that, else I can start putting together something we can add a test for your parameter to, but I'm not sure when I'll be able to get to that.

jamesoff commented:
All the tests passed here, not sure what's up with your local failures. If you want to fix the HTML one (by making that directory) and still get the other ones, feel free to share the full log and I'll take a look. I think `poetry run pytest -v tests` should generate the output, but if not add another `v` :)

kz159 commented:
Hi, thanks for ideas, i've added ground for future tests for this monitor and added couple of words in docs, there are still missing info about certfile and keyfile btw

jamesoff commented:
Thanks again for this, and apologies for sitting on it for so long.
